File: androidApp/src/main/java/se/fransman/tgapp/composables/navigation/ScreenPicker.kt
package se.fransman.tgapp.composables.navigation

import androidx.compose.runtime.Composable
import se.fransman.tg.shared.viewmodel.Navigation
import se.fransman.tg.shared.viewmodel.ScreenIdentifier
import se.fransman.tg.shared.viewmodel.screens.Screen.*
import se.fransman.tgapp.composables.screens.HomeScreen


@Composable
fun Navigation.ScreenPicker(
  screenIdentifier: ScreenIdentifier
) {

  when (screenIdentifier.screen) {

    Home ->
      HomeScreen(
        homeState = stateProvider.get(screenIdentifier),
        // onListItemClick = { navigate(CountryDetail, CountryDetailParams(countryName = it)) },
      )

  }
}

File: iosApp/iosApp/Views/navigation/ScreenPicker.swift
import SwiftUI
import shared

extension Navigation {
    @ViewBuilder func screenPicker(_ sId: ScreenIdentifier) -> some View {
        VStack {
            switch sId.screen {

            case .home:
                HomeScreen(homeState: self.stateProvider.getToCast(screenIdentifier: sId) as! HomeState)
            default:
                EmptyView()
            }

        }
    }
}
